infra/terraform/README.md

```diff
@@ -0,0 +1,249 @@
+
+   AWS Single Sign-On (SSO) allows you to manage access to your AWS resources using your organization's SSO provider. Here's how to set it up:
+
+   1. **Configure AWS SSO:**
+      - Sign in to the [AWS SSO Console](https://console.aws.amazon.com/singlesignon/).
+      - Follow the [AWS documentation](https://docs.aws.amazon.com/singlesignon/latest/userguide/getting-started.html) to set up SSO with your identity provider.
+
+   2. **Create IAM Roles for SSO:**
+      - Go to the [IAM Console](https://console.aws.amazon.com/iam/home).
+      - Create a new role and select **AWS SSO** as the trusted entity.
+      - Assign permissions that are necessary for Terraform operations. Common permissions include `AmazonEC2FullAccess`, `AmazonVPCFullAccess`, `AmazonS3FullAccess`, and `IAMFullAccess`.
+
+   3. **Assign Roles to Users:**
+      - Go back to the AWS SSO Console.
+      - Assign the newly created IAM roles to users or groups within your SSO setup.
+
+   4. **Configure Terraform to Use SSO:**
+      - Update your Terraform provider configuration to use the `aws` profile that corresponds to your SSO setup in `provider.tf`:
+      ```hcl
+      provider "aws" {
+         profile = "your-sso-profile"
+         region  = "us-west-2"
+      }
+      ```
+
+   5. **Authenticate Using SSO:**
+      - Use the AWS CLI to authenticate with SSO before running Terraform commands:
+      ```bash
+      aws sso login --profile your-sso-profile
+      ```
+
+   ### Using IAM User Role Assumption
+
+   For environments where SSO is not used, you can set up IAM users with minimal permissions and grant them the ability to assume a more powerful role.
+
+   1. **Create a Role with Necessary Permissions:**
+      - Go to the [IAM Console](https://console.aws.amazon.com/iam/home).
+      - Create a new IAM role with the permissions required for Terraform operations `AmazonEC2FullAccess`, `AmazonS3FullAccess`, `IAMFullAccess`, `AmazonVPCFullAccess`.
+
+   2. **Create a Trust Policy for the Role:**
+      - Add a trust policy to allow specific IAM users or roles to assume this role. Example trust policy:
+      ```json
+      {
+         "Version": "2012-10-17",
+         "Statement": [
+            {
+            "Effect": "Allow",
+            "Principal": {
+               "AWS": [
+                  "arn:aws:iam::123456789012:user/terraform-user"
+               ]
+            },
+            "Action": "sts:AssumeRole"
+            }
+         ]
+      }
+      ```
+
+   3. **Create IAM Users with Limited Permissions:**
+      - Create IAM users who will
+
+   use Terraform.
+      - Grant these users only the permission to assume the previously created role. Example policy for the user:
+      ```json
+      {
+         "Version": "2012-10-17",
+         "Statement": [
+            {
+            "Effect": "Allow",
+            "Action": "sts:AssumeRole",
+            "Resource": "arn:aws:iam::123456789012:role/terraform-role"
```

Comment:
The examples you used everywhere uses `123456789012` as the account number. I'd replace it with a placeholder `<account-number>` or something.